src/CarbonAware.DataSources/CarbonAware.DataSources.ElectricityMapsFree/test/Client/ElectricityMapsFreeClientTests.cs

```diff
@@ -0,0 +1,169 @@
+        this.Configuration = new ElectricityMapsFreeClientConfiguration()
+        {
+            Token = token,
+            BaseUrl = baseUrl,
+        };
+
+        // Act & Assert
+        Assert.Throws<ConfigurationException>(() => new ElectricityMapsFreeClient(this.HttpClientFactory, this.Options.Object, this.Log.Object));
+    }
+
+    [Test]
+    public void AllPublicMethods_DoNotSwallowBadProxyExceptions()
+    {
+        // Arrange
+        var mockHttpClientFactory = Mock.Of<IHttpClientFactory>();
+        var mockHandler = new Mock<HttpClientHandler>();
+        this.Configuration = GetValidConfiguration();
+
+        // A bad proxy will throw HttpRequestException when used so we mock that here.
+        mockHandler.Protected()
+            .Setup<Task<HttpResponseMessage>>(
+                "SendAsync",
+                ItExpr.IsAny<HttpRequestMessage>(),
+                ItExpr.IsAny<CancellationToken>()
+            )
+            .ThrowsAsync(new HttpRequestException());
+
+        Mock.Get(mockHttpClientFactory)
+            .Setup(h => h.CreateClient(It.IsAny<string>()))
+            .Returns(new HttpClient(mockHandler.Object));
+
+        var client = new ElectricityMapsFreeClient(mockHttpClientFactory, this.Options.Object, this.Log.Object);
+
+        // Act & Assert
+        Assert.ThrowsAsync<HttpRequestException>(async () => await client.GetCurrentEmissionsAsync(TestCountryCode));
+        Assert.ThrowsAsync<HttpRequestException>(async () => await client.GetCurrentEmissionsAsync(TestLatitude, TestLongitude));
+    }
+
+    [Test]
+    public void AllPublicMethods_ThrowJsonException_WhenBadJsonIsReturned()
+    {
+        // Arrange
+        this.Configuration = GetValidConfiguration();
+        AddHandler_RequestResponse(r =>
+        {
+            return r.RequestUri!.ToString().Contains(Paths.Latest) && r.Method == HttpMethod.Get;
+        }, System.Net.HttpStatusCode.OK, "This is bad json");
+
+        var client = new ElectricityMapsFreeClient(this.HttpClientFactory, this.Options.Object, this.Log.Object);
+
+        // Act & Assert
+        Assert.ThrowsAsync<JsonException>(async () => await client.GetCurrentEmissionsAsync(TestLatitude, TestLongitude));
+    }
+
+    [Test]
+    public async Task GetLatestCarbonIntensityAsync_DeserializesExpectedResponse()
+    {
+        // Arrange
+        this.Configuration = GetValidConfiguration();
+        AddHandler_RequestResponse(r =>
+        {
+            return r.RequestUri!.ToString().Contains(Paths.Latest) && r.Method == HttpMethod.Get;
+        }, System.Net.HttpStatusCode.OK, TestData.GetLatestCarbonIntensityDataJsonString());
+
+        var client = new ElectricityMapsFreeClient(this.HttpClientFactory, this.Options.Object, this.Log.Object);
+
+        // Act
+        var latestData = await client.GetCurrentEmissionsAsync(TestLatitude, TestLongitude);
+        var dataPoint = latestData?.Data;
+
+        // Assert
+        Assert.That(latestData, Is.Not.Null);
```

Comment:
Move this assert before line 127. If `latestData` is null, NPE will be thrown and this assert will not be executed.